src/Sylius/Bundle/ShopBundle/spec/Router/LocaleStrippingRouterSpec.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ShopBundle\Router;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Locale\Context\LocaleContextInterface;
+use Symfony\Component\HttpKernel\CacheWarmer\WarmableInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+final class LocaleStrippingRouterSpec extends ObjectBehavior
+{
+    function let(RouterInterface $decoratedRouter, LocaleContextInterface $localeContext): void
+    {
+        $this->beConstructedWith($decoratedRouter, $localeContext);
+    }
+
+    function it_is_a_symfony_router(): void
+    {
+        $this->shouldImplement(RouterInterface::class);
+    }
+
+    function it_is_warmable(): void
+    {
+        $this->shouldImplement(WarmableInterface::class);
+
+        $this->warmUp('/cache/dir');
+    }
+
+    function it_strips_locale_from_the_generated_url_if_locale_is_the_same_as_the_one_from_context(
+        RouterInterface $decoratedRouter,
+        LocaleContextInterface $localeContext
+    ): void {
+        $localeContext->getLocaleCode()->willReturn('pl_PL');
+
+        $decoratedRouter
+            ->generate('route_name', [], UrlGeneratorInterface::ABSOLUTE_PATH)
+            ->willReturn('http://generated.url/?_locale=pl_PL')
```

Comment:
Can we test for each case? It should be as easy as adding here two more examples and calling generate two more times

Reply:
Done! 👌